apps/src/labs/labRedux.ts

```diff
@@ -33,7 +33,10 @@ import {
 } from './progress/ProgressManager';
 
 export interface LabState {
-  // If we are currently loading a lab.
+  // If we are currently loading common data for a project or level. Should only be used internally
+  // by this Redux file.
+  isLoadingProjectOrLevel: boolean;
+  // If the lab is loading. Can be updated by lab-specific components.
   isLoading: boolean;
```

Comment:
I don't love having these two separate flags, but I was worried that if labs are also updating the same `isLoading` flag, we'd get into weird states where the lab may say that loading is done while the redux thunks are still in flight, or vice versa. Curious how y'all feel

Reply:
I think as long as it's isolated to internal use only it's ok, and I agree that we could get into weird states without it. We should keep an eye on this going forward and if we find we need another flag maybe we create some sort of loading manager that handles parsing these load states? Or extract the load state data to a separate redux slice?